src/raftstore/store/store.rs

```diff
@@ -1876,24 +1879,46 @@ impl<T: Transport, C: PdClient> Store<T, C> {
     }
 
     fn on_compact_check_tick(&mut self, event_loop: &mut EventLoop<Self>) {
-        for peer in self.region_peers.values_mut() {
-            if peer.delete_keys_hint < self.cfg.region_compact_delete_keys_count {
-                continue;
-            }
-            for &cf in &[CF_DEFAULT, CF_WRITE] {
-                let task = CompactTask {
-                    cf_name: String::from(cf),
-                    start_key: Some(keys::enc_start_key(peer.region())),
-                    end_key: Some(keys::enc_end_key(peer.region())),
-                };
-                if let Err(e) = self.compact_worker.schedule(task) {
-                    error!("{} failed to schedule compact task: {}", self.tag, e);
+        if self.compact_worker.is_busy() {
+            info!("compact worker is busy, check space redundancy next time");
```

Comment:
You can return early to reduce indent.

Reply:
After refactor, there are 3 branches in this function, we must register tick 3 times if return early to reduce indent.